File: tests/data/repo_chunking/sample.c
#include <stdio.h>
#include <math.h>

double circle_area(double radius) {
    return M_PI * radius * radius;
}

double circle_circumference(double radius) {
    return 2 * M_PI * radius;
}

int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
    printf("Hello, World!\n");
    return 0;
}

File: tests/data/repo_chunking/sample.py
def fibonacci(n):
    """Calculate the nth Fibonacci number using recursion."""
    if n <= 1:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)


class MathUtils:
    """Utility class for mathematical operations."""

    def __init__(self):
        self.pi = 3.14159

    def circle_area(self, radius):
        """Calculate the area of a circle."""
        return self.pi * radius**2

    def circle_circumference(self, radius):
        """Calculate the circumference of a circle."""
        return 2 * self.pi * radius


def hello():
    """A simple hello function."""
    return "Hello, World!"
